### PR TITLE
add functional requirements.txt and use cases

### PR DESCRIPTION
**Pull request by Depermitto**
Add polish version of:

# Functional Requirements:

- The user will be able to create their own account
- The user will be able to log in to their account, change their password, and delete it
- The user will be able to send and receive files
- The user will be able to share files with another user
- The user will be able to create private files that are invisible to other users
- The user will be able to categorize files and tag them
- The user will be able to search for files in the database (to which they have access)
    - by name
    - by type (.pdf, .zip, .txt)
    - by content (in the case of text files)
    - by size
    - by modification date
    - by creation date
- The user will be able to perform basic operations on files, such as:
    - deleting a file
    - changing the file name
    - changing the file location on disk
- The user will be able to create folders in a tree structure and place files in them
- The user will be able to encrypt files/folders with a password
- The user will be able to view their own audit trail (digital activity trace, i.e., how many files were downloaded, when they were downloaded)
- The user will be able to export the state of their disk (the state includes the disk structure, encrypted files, tags, and categories)
- The user will be able to import the state of their disk from a file

# Use Cases:

- Creating an account to be able to use the platform
- Sending files
- Receiving files
- Sharing files with another user
- Changing the name of a file
- Deleting a file
- Encrypting folders/files
- Searching for files by metadata or content
- Categorizing/tagging files
- Viewing their own audit trail
- Exporting/Importing the state of the disk (e.g., to not lose data during server changes)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,6 +39,46 @@ Na dowolnym systemie wspierającym konteneryzację Docker można uruchomić cał
...
+- Użytkownik będzie w stanie wykonywać podstawowe operacje na plikach, takie jak:
+    - usunięcie pliku
+    - zmienianie nazwy
+    - zmiana miejsca w dysku
+- Użytkownik będzie w stanie tworzyć foldery w strukturze drzewiastej i umieszczać tam pliki
+- Użytkownik będzie w stanie szyfrować pliki/foldery hasłem
+- Użytkownik będzie w stanie zobaczyć własną ewidencję (ślad cyfrowego aktywności tzn. ile było pobrać, kiedy pobrał)
+- Użytkownik będzie w stanie wyeksportować stan swojego dysku (stan obejmuje strukturę dysku, zaszyfrowane pliki, tagi
+  oraz kategorie)
+- Użytkownik będzie w stanie zaimportować stan dysku z pliku
```
Comment (maksnowak) [inline README.md:66]: I'm not sure if we're gonna be able to add it in such a short period of time

File: README.md [hunk 2]
```diff
@@ -39,6 +39,46 @@ Na dowolnym systemie wspierającym konteneryzację Docker można uruchomić cał
...
+    - po rozmiarze
+    - po dacie modyfikacji
+    - po dacie utworzenia
+- Użytkownik będzie w stanie wykonywać podstawowe operacje na plikach, takie jak:
+    - usunięcie pliku
+    - zmienianie nazwy
+    - zmiana miejsca w dysku
+- Użytkownik będzie w stanie tworzyć foldery w strukturze drzewiastej i umieszczać tam pliki
+- Użytkownik będzie w stanie szyfrować pliki/foldery hasłem
+- Użytkownik będzie w stanie zobaczyć własną ewidencję (ślad cyfrowego aktywności tzn. ile było pobrać, kiedy pobrał)
```
Comment (maksnowak) [inline README.md:63]: pobrać -> pobrań

File: README.md [hunk 3]
```diff
@@ -39,6 +39,46 @@ Na dowolnym systemie wspierającym konteneryzację Docker można uruchomić cał
...
+- Wysyłanie plików
+- Pobieranie plików
+- Udostępnienie plików innemu użytkownikowi
+- Zmiana nazwy pliku
+- Usunięcie pliku
+- Szyfrowanie folderów/plików
+- Wyszukiwanie plików po metadanych lub zawartości
+- Kategoryzacja/tagowanie plików
+- Podejrzenie własnej ewidencji
+- Wyeksportowanie/Zaimportowanie stanu dysku (np. żeby nie stracić danych podczas zmiany serwera)
```
Comment (maksnowak) [inline README.md:80]: See previous comment

**Review by maksnowak — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Depermitto) [reply README.md:66]: what's this then [check bullet point no. 3](https://github.com/maksnowak/FileSharingSystem?tab=readme-ov-ile#opis-projektu)

File: README.md (hunk 3 above)
Comment (Depermitto) [reply README.md:80]: See previous comment

File: README.md (hunk 1 above)
Comment (maksnowak) [reply README.md:66]: mea culpa

File: README.md (hunk 2 above)
Comment (Depermitto) [reply README.md]: fixed in 9eef1ff23518

**Review by maksnowak — APPROVED** [review]

**Pull request merged**